src/navigator/main-stack.tsx

```diff
@@ -47,14 +48,16 @@ const DATA_REFRESH_INTERVAL = 60 * 1000;
 const EXCHANGE_RATE_REFRESH_INTERVAL = 5 * 60 * 1000;
 
 export const MainStackScreen = () => {
+  const [isBeaconConnected, setIsBeaconConnected] = useState(false);
   const dispatch = useDispatch();
   const isAuthorised = useIsAuthorisedSelector();
   const selectedAccount = useSelectedAccountSelector();
   const styleScreenOptions = useStackNavigatorStyleOptions();
+  BeaconHandler.isBeaconConnectedHandler().then(isConnected => setIsBeaconConnected(isConnected));
```

Comment:
wrong data change propagation flow.  
move `isConnected` state into `useBeaconHandler`. 
merge `useDeepLink` with `useBeaconHandler` as we are handling only Beacons links